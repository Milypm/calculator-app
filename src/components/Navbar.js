import React from 'react';
import { Link } from 'react-router-dom';
import '../style/style.css';

const Navbar = () => (
  <div className="header-navbar">
    <div className="header-icon">
      <h1>Math Magicians</h1>
      <i className="fas fa-hat-wizard" />
    </div>
    <ul className="nav-ul">
      <li className="li-separator"><Link to="/">Home</Link></li>
      <li className="li-separator"><Link to="/calculator">Calculator</Link></li>
      <li><Link to="/quote-of-the-day">Quote of the Day</Link></li>
    </ul>
  </div>
);

export default Navbar;
