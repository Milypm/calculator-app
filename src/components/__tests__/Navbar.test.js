import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { BrowserRouter } from 'react-router-dom';
import Navbar from '../Navbar';

const renderNavbar = () => {
  render(
    <BrowserRouter>
      <Navbar />
    </BrowserRouter>
  );
};

describe('Navbar component', () => {
  test('renders Navbar component', () => {
    render(
      <BrowserRouter>
        <Navbar />
      </BrowserRouter>
    );
  });
  test('displays specific text on the screen', () => {
    renderNavbar();
    expect(screen.getByText('Home')).toBeInTheDocument();
    expect(screen.getByText(/Quote of the Day/)).toBeInTheDocument();
  });
  test('displays a list of clickable Links to go from one view to another', () => {
    renderNavbar();
    expect(screen.getByRole('list')).toBeInTheDocument();
  });
});
