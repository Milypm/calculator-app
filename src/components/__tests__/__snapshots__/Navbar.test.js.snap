// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar component renders Navbar component 1`] = `
<div
  className="header-navbar"
>
  <div
    className="header-icon"
  >
    <h1>
      Math Magicians
    </h1>
    <i
      className="fas fa-hat-wizard"
    />
  </div>
  <ul
    className="nav-ul"
  >
    <li
      className="li-separator"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
    </li>
    <li
      className="li-separator li-calc"
    >
      <a
        href="/calculator"
        onClick={[Function]}
      >
        Calculator
      </a>
    </li>
    <li>
      <a
        href="/quote-of-the-day"
        onClick={[Function]}
      >
        Quote of the Day
      </a>
    </li>
  </ul>
</div>
`;
